android/java/org/chromium/chrome/browser/brave_news/CardBuilderFeedCard.java

```diff
@@ -169,6 +180,214 @@ private GradientDrawable roundedBackground() {
+                    openUrlInSameTabAndSavePosition(BraveActivity.BRAVE_REWARDS_SETTINGS_URL);
+                }
+            });
+
+            rowTableParams.setMargins(50, 0, 50, 0);
+            rowTableParams.width = TableLayout.LayoutParams.MATCH_PARENT;
+            rowTableParams.height = TableLayout.LayoutParams.WRAP_CONTENT;
+            row1.setLayoutParams(rowTableParams);
+            rowTableParams.bottomMargin = 3 * MARGIN_VERTICAL;
+            row2.setLayoutParams(rowTableParams);
+            tableLayoutTopNews.addView(row1);
+            tableLayoutTopNews.addView(row2);
+
+            row1.addView(adTitle);
+            row2.addView(adDesc);
+            row2.addView(adButton);
+
+            adItemsParams = new TableRow.LayoutParams(
+                    TableRow.LayoutParams.MATCH_PARENT, TableRow.LayoutParams.WRAP_CONTENT, 1.0f);
+            adTitle.setTextSize(17);
+            adItemsParams.bottomMargin = 2 * MARGIN_VERTICAL;
+            adTitle.setTypeface(null, Typeface.BOLD);
+            adTitle.setMaxLines(3);
+            adTitle.setTextColor(mActivity.getResources().getColor(R.color.news_text_color));
+            adTitle.setEllipsize(TextUtils.TruncateAt.END);
+            adTitle.setLayoutParams(adItemsParams);
+            adTitle.setText(adData.title);
+
+            adItemsParams = new TableRow.LayoutParams(
+                    TableRow.LayoutParams.WRAP_CONTENT, TableRow.LayoutParams.MATCH_PARENT);
+            adItemsParams.gravity = Gravity.CENTER_VERTICAL;
+            adDesc.setGravity(Gravity.CENTER_VERTICAL);
+            adItemsParams.weight = 1;
+            adItemsParams.width = 0;
+            adDesc.setLayoutParams(adItemsParams);
+
+            adDesc.setTextColor(mActivity.getResources().getColor(R.color.news_time_color));
+            adDesc.setTextSize(12);
+            adDesc.setText(adData.description);
+
+            adButtonParams.width = TableRow.LayoutParams.WRAP_CONTENT;
+            adButtonParams.height = 80; // TableRow.LayoutParams.WRAP_CONTENT;
+            adButton.setPadding(30, 0, 30, 0);
+            adButton.setTextSize(13);
+            adButton.setAllCaps(false);
+            GradientDrawable adButtonBG = new GradientDrawable();
+            adButtonBG.setColor(mActivity.getResources().getColor(android.R.color.transparent));
+            adButtonBG.setCornerRadius(55);
+            adButtonBG.setStroke(1, mActivity.getResources().getColor(R.color.news_text_color));
+            adButton.setBackground(adButtonBG);
+            adButton.setText(adData.ctaText);
+            adButton.setTextColor(mActivity.getResources().getColor(R.color.news_text_color));
+            adButton.setLayoutParams(adButtonParams);
+            adButton.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    mBraveNewsController.onDisplayAdVisit(adData.uuid, adData.creativeInstanceId);
```

Comment:
here and below there are 4 equal `OnClickListener`s with more than one line.

Maybe it worth to make it be invoked from the one place to avoid accidental mismatch after future changes?
Like 
```
+            Runnable commonOnClickListener = () -> {
+                mBraveNewsController.onDisplayAdVisit(adData.uuid, adData.creativeInstanceId);
+                openUrlInSameTabAndSavePosition(adData.targetUrl.url);
+            };
+
             adButton.setOnClickListener(new View.OnClickListener() {
                 @Override
                 public void onClick(View v) {
-                    mBraveNewsController.onDisplayAdVisit(adData.uuid, adData.creativeInstanceId);
-                    openUrlInSameTabAndSavePosition(adData.targetUrl.url);
+                    commonOnClickListener.run();
                 }
```

Reply:
thanks, your change 
```
           View.OnClickListener listener = v -> {
                openUrlAndSaveEvent(adData);
            };

            adButton.setOnClickListener(listener);
```
even better than my proposition